sr_error_reporter/src/UnderactuationErrorReporter.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+* Copyright 2021 Shadow Robot Company Ltd.
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the Free
+* Software Foundation version 2 of the License.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "eigen_conversions/eigen_msg.h"
+#include "std_msgs/String.h"
+#include "sr_error_reporter/UnderactuationError.h"
+#include "sr_error_reporter/UnderactuationErrorReporter.hpp"
+
+UnderactuationErrorReporter::UnderactuationErrorReporter(ros::NodeHandle& node_handle)
+  : node_handle_(node_handle)
+{
+  joints_subsriber_ = node_handle.subscribe("/joint_states", 1,
+    &UnderactuationErrorReporter::joints_callback, this, ros::TransportHints().tcpNoDelay());
+  trajectory_subsriber_left_ = node_handle.subscribe("/lh_trajectory_controller/command", 1,
+    &UnderactuationErrorReporter::trajectory_callback, this, ros::TransportHints().tcpNoDelay());
+  trajectory_subsriber_right_ = node_handle.subscribe("/rh_trajectory_controller/command", 1,
+    &UnderactuationErrorReporter::trajectory_callback, this, ros::TransportHints().tcpNoDelay());
+
+  robot_model_loader_.reset(new robot_model_loader::RobotModelLoader("robot_description"));
+  robot_state_.reset(new robot_state::RobotState(robot_model_loader_->getModel()));
+  robot_state_->setToDefaultValues();
+}
+
+ros::Publisher UnderactuationErrorReporter::get_or_create_publisher(std::string link_name)
+{
+  auto iterator = error_publishers_.find(link_name);
+  if (iterator != error_publishers_.end())
+  {
+    return iterator->second;
+  }
+  ros::Publisher publister = node_handle_.advertise<sr_error_reporter::UnderactuationError>(
```

Comment:
publisher typo

Reply:
Renamed in bf3029a16310672b6159f2cfc6a2e764489858e1.